app/src/test/java/com/nervousfish/nervousfish/modules/pairing/RhythmHelperTest.java

```diff
@@ -0,0 +1,178 @@
+        masterData.add(new SingleTap(new Timestamp(0L)));
+        final RhythmHelper helper = new RhythmHelper(masterData);
+
+        List<ATapData> slaveData = new ArrayList<>();
+        slaveData.add(new SingleTap(new Timestamp(1000000L)));
+        helper.setSlaveData(slaveData);
+        assertTrue(helper.isMatch(2, 10));
+    }
+
+    @Test
+    public void testFiveMatchByAdd() throws Exception {
+        List<ATapData> masterData = new ArrayList<>();
+        masterData.add(new SingleTap(new Timestamp(0L)));
+        masterData.add(new SingleTap(new Timestamp(20L)));
+        masterData.add(new SingleTap(new Timestamp(40L)));
+        masterData.add(new SingleTap(new Timestamp(60L)));
+        masterData.add(new SingleTap(new Timestamp(80L)));
+        final RhythmHelper helper = new RhythmHelper(masterData);
+
+        helper.addSlaveData(new SingleTap(new Timestamp(1000000L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000018L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000036L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000065L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000081L)));
+        assertTrue(helper.isMatch(2, 20));
+    }
+
+    @Test
+    public void testFiveNoMatchByAdd() throws Exception {
+        List<ATapData> masterData = new ArrayList<>();
+        masterData.add(new SingleTap(new Timestamp(0L)));
+        masterData.add(new SingleTap(new Timestamp(20L)));
+        masterData.add(new SingleTap(new Timestamp(40L)));
+        masterData.add(new SingleTap(new Timestamp(60L)));
+        masterData.add(new SingleTap(new Timestamp(80L)));
+        final RhythmHelper helper = new RhythmHelper(masterData);
+
+        helper.addSlaveData(new SingleTap(new Timestamp(1000000L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000018L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000036L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000065L)));
+        helper.addSlaveData(new SingleTap(new Timestamp(1000081L)));
+        assertFalse(helper.isMatch(2, 2));
+    }
+
+    @Test
+    public void testFiveMatchBySet() throws Exception {
+        List<ATapData> masterData = new ArrayList<>();
+        masterData.add(new SingleTap(new Timestamp(0L)));
+        masterData.add(new SingleTap(new Timestamp(20L)));
+        masterData.add(new SingleTap(new Timestamp(40L)));
+        masterData.add(new SingleTap(new Timestamp(60L)));
+        masterData.add(new SingleTap(new Timestamp(80L)));
+        final RhythmHelper helper = new RhythmHelper(masterData);
+
+        List<ATapData> slaveData = new ArrayList<>();
+        slaveData.add(new SingleTap(new Timestamp(1000000L)));
+        slaveData.add(new SingleTap(new Timestamp(1000018L)));
+        slaveData.add(new SingleTap(new Timestamp(1000036L)));
+        slaveData.add(new SingleTap(new Timestamp(1000065L)));
+        slaveData.add(new SingleTap(new Timestamp(1000081L)));
```

Comment:
Because the slave takes a veeery long time to respond to the taps and I want to be sure that the algorithm still recognizes the rhythm even he waits for a very long time